internal/resources/resource_host.go

```diff
@@ -776,31 +785,49 @@ func getNetworkIDs(d *schema.ResourceData, p *configuration.Config, host *rest.H
 	return netIds, nil
 }
 
-// getNetworkDefRoute returns the network for default route specified in the request or
-// the current value maintained in the state.
-func getNetworkDefRoute(d *schema.ResourceData, p *configuration.Config, host *rest.Host) (nDefRoute string, err error) {
-	nDefRoute, ok := d.Get(hNetForDefaultRoute).(string)
-	if !ok {
-		// network for default route is optional
-		// return the current value
-		return host.NetworkForDefaultRoute, nil
+// getNetworkID returns the network ID specified in the request.
+func getNetworkID(p *configuration.Config, hostNets []string, locationID, net string) (string, error) {
+	if net == "" {
+		return net, nil
```

Comment:
If the caller didn't specify a network to get the ID for, is it really not an error (error = nil)?

Reply:
That is done because the untagged network is optional